src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,75 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  todo: Todo,
+  setTodo: (obj: Todo | null) => void,
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, setTodo }) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then(setSelectedUser);
```

Comment:
Let’s use `async await` all over the code. It gives you a simpler way to work with asynchronous promise-based code